src/services/expenses.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+let expenses = [];
+
+const initialExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (userId, category, from, to) => {
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.userId === Number(userId));
+  }
+
+  if (category) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.category === category);
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.spentAt > from);
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.spentAt < to);
+  }
+
+  return filteredExpenses;
+};
+
+const getExpensesById = (expenseId) => {
+  const foundExpense = expenses
+    .find(expense => expense.id === Number(expenseId));
+
+  return foundExpense || null;
+};
+
+const addExpense = (userId, spentAt, title, amount, category, note) => {
+  let newExpense = {};
+
+  if (!expenses.length) {
+    newExpense = {
+      id: 0,
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    };
+  } else {
+    const maxId = Math.max(...expenses.map(expense => expense.id));
+
+    newExpense = {
+      id: maxId + 1,
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    };
+  }
```

Comment:
all that changes is id, you repeat yourself on everything else, please use ternary variable for your id